### PR TITLE
ENH: add 'alternative' and 'exact' keywords to `stats.ansari`

### LINKED ISSUE #688: Statistics Review: ansari (Trac #161)
_Original ticket http://projects.scipy.org/scipy/ticket/161 on 2006-04-02 by @rkern, assigned to @rkern._

The function `ansari` in file source:trunk/Lib/stats/morestats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

### PR DESCRIPTION
**Pull request by aeklant**
closes gh-688

### CODE REVIEW

**Comment by aeklant** [comment]
@rgommers ping

I wanted to push an initial commit to review first just to make sure I understood the idea behind 'exact' and 'alternative'. I will add regression tests once I have some feedback about this.

File: scipy/stats/morestats.py [hunk 1]
```diff
@@ -1474,6 +1478,10 @@ def ansari(x, y):
            methods.  3rd ed. Chapman and Hall/CRC. 2001.  Section 5.8.2.
 
     """
+    if alternative not in ('two-sided', 'less', 'greater'):
+        raise ValueError("alternative not recognized\n"
```
Comment (rgommers) [inline scipy/stats/morestats.py]: no need for ``\n`` here, you can replace it with ``, `` to keep the output on a single line.

File: scipy/stats/morestats.py [hunk 2]
```diff
@@ -1464,9 +1469,8 @@ def ansari(x, y):
 
     Notes
     -----
-    The p-value given is exact when the sample sizes are both less than
-    55 and there are no ties, otherwise a normal approximation for the
-    p-value is used.
+    Exact test should be used when the sample sizes are both less than 55 and
```
Comment (rgommers) [inline scipy/stats/morestats.py]: "should be" is maybe not the right phrasing, this is what ``'auto'`` does but that's not necessarily much better or worse than putting the cutoff at 50 or 60.

**Comment by rgommers** [comment]
The ``exact`` keyword looks good to me. Haven't checked ``alternative`` yet.

**Comment by rgommers** [comment]
The implementation of ``alternative`` here doesn't look right. The alternative hypotheses are:

- ratio of scales is not equal to 1
- ratio of scales is less than 1
- ratio of scales is larger than 1

If you have two inputs ``x1`` and ``x2``, then if ``x1`` has larger variance than ``x2`` the p-value for ``'less'`` has to be smaller than the p-value for ``'greater'``. In the current implementation they are the same.

I'll comment on gh-688 in more detail.

**Comment by mdhaber** [comment]
Closing in favor of gh-13650

**Pull request closed without merging**